internal/services/inference/deployment.go

```diff
@@ -0,0 +1,271 @@
+			},
+			"min_size": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The minimum size of the pool",
+			},
+			"max_size": {
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "The maximum size of the pool",
+			},
+			"size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The size of the pool",
+			},
+			"status": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The status of the deployment",
+			},
+			"endpoint_public_url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint public URL",
+			},
+			"endpoint_private_url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint private URL",
+			},
+			"endpoint_public_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint public ID",
+			},
+			"endpoint_private_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint private ID",
+			},
+			"endpoints": {
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: "List of endpoints",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"public_endpoint": {
+							Type:        schema.TypeBool,
+							Description: "Set the endpoint as public",
+							Optional:    true,
+						},
+						"private_endpoint": {
+							Type:        schema.TypeString,
+							Description: "The id of the private network",
+							Optional:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func ResourceDeploymentCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	api, region, err := NewAPIWithRegion(d, m)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	endpoint := inference.EndpointSpec{
+		Public:         nil,
+		PrivateNetwork: nil,
+		DisableAuth:    false,
+	}
+
+	req := &inference.CreateDeploymentRequest{
+		Region:    region,
+		ProjectID: d.Get("project_id").(string),
+		Name:      d.Get("name").(string),
+		NodeType:  d.Get("node_type").(string),
+		ModelName: d.Get("model_name").(string),
+		Tags:      types.ExpandStrings(d.Get("tags")),
+	}
+
+	if _, isEndpoint := d.GetOk("endpoints"); isEndpoint {
+		if publicEndpoint := d.Get("endpoints.0.public_endpoint"); publicEndpoint != nil && publicEndpoint.(bool) {
+			endpoint.Public = &inference.EndpointSpecPublic{}
+		}
+		if privateEndpoint := d.Get("endpoints.0.private_endpoint"); privateEndpoint != "" {
+			endpoint.PrivateNetwork = &inference.EndpointSpecPrivateNetwork{
+				PrivateNetworkID: privateEndpoint.(string),
+			}
+		}
+	}
+
+	if maxSize, ok := d.GetOk("max_size"); ok {
+		req.MaxSize = scw.Uint32Ptr(uint32(maxSize.(int)))
+	}
+
+	if minSize, ok := d.GetOk("min_size"); ok {
+		req.MaxSize = scw.Uint32Ptr(uint32(minSize.(int)))
+	}
+
+	req.Endpoints = []*inference.EndpointSpec{&endpoint}
```

Comment:
Move the code that is referring to endpoints closer together to reduce cognitive load